isis/src/control/objs/ControlNet/ControlNet.cpp

```diff
@@ -369,16 +357,20 @@ namespace Isis {
     // make sure there is a node for every measure
     for (int i = 0; i < point->GetNumMeasures(); i++) {
       QString sn = point->GetMeasure(i)->GetCubeSerialNumber();
-      if (!cubeGraphNodes->contains(sn)) {
-        cubeGraphNodes->insert(sn, new ControlCubeGraphNode(sn));
+
+      // if the graph doesn't have the sn
+      if (!m_vertexMap.contains(sn)) {
+        ImageVertex newVertex = boost::add_vertex(m_controlGraph); 
+        m_controlGraph[newVertex].serial = sn; 
```

Comment:
Based on the documentation, you can combine this line with the one above it using add_vertex(properties, graph). It's probably an incredibly small optimization though.

Reply:
:+1: